.circleci/config.yml

```diff
@@ -87,49 +107,84 @@ jobs:
+          working_directory: ~/expo
       - save_yarn_cache
       # Add back linting once we get ESLint or TSLint set up
-      - jest:
-          dir: ~/expo/packages/expo
+      - yarn:
+          command: jest --maxWorkers 1
+          working_directory: ~/expo/packages/expo
 
   babel_preset:
     executor: js
     steps:
-      - checkout
-      - set_up_environment
+      - setup
+      - submodule_update
       - restore_yarn_cache
-      - install_workspace_dependencies
+      - yarn:
+          command: install
+          working_directory: ~/expo
       - save_yarn_cache
-      - lint:
-          dir: ~/expo/packages/babel-preset-expo
-      - jest:
-          dir: ~/expo/packages/babel-preset-expo
+      - yarn:
+          command: lint --max-warnings 0
+          working_directory: ~/expo/packages/babel-preset-expo
+      - yarn:
+          command: jest --maxWorkers 1
+          working_directory: ~/expo/packages/babel-preset-expo
 
   home:
     executor: js
     steps:
-      - checkout
-      - set_up_environment
+      - setup
+      - submodule_update
       - restore_yarn_cache
-      - install_workspace_dependencies
+      - yarn:
+          command: install
+          working_directory: ~/expo
       - save_yarn_cache
-      - jest:
-          dir: ~/expo/home
+      - yarn:
+          command: jest --maxWorkers 1
+          working_directory: ~/expo/home
 
   client_ios:
     executor: mac
     steps:
-      - checkout
-      - set_up_environment
+      - install_nix
+      - setup
       - fetch_cocoapods_specs
-      - run:
+      - yarn:
+          command: install
           working_directory: ~/project/tools-public
-          command: yarn install
-      - run:
-          working_directory: ~/project/tools-public
-          command: PATH=$(yarn bin):$PATH ./generate-files-ios.sh
+      - run: nix run nixpkgs.nodejs-8_x -c ~/project/tools-public/generate-files-ios.sh
       - run: fastlane ios create_simulator_build
+
+  client_android:
+    executor: android
+    steps:
+      - install_nix
+      - setup
+      - yarn:
+          command: install
+          working_directory: ~/expo/tools-public
+      - run: nix-env -f . -iA nodejs-8_x # For some reason `nix run` doesn't work
+      - run: ~/expo/tools-public/generate-dynamic-macros-android.sh
+      - restore_gradle_cache
+      - run:
+          working_directory: ~/expo/android
+          command: ./gradlew :app:preBuild
+      - save_gradle_cache
+      - run:
+          working_directory: ~/expo/android
+          command: ./gradlew :app:assembleProdMinSdkProdKernelRelease
+      - run: echo 'export PATH="/opt/android/sdk/build-tools/27.0.3:$PATH"' >> $BASH_ENV # should match buildToolsVersion in android/app/build.gradle
+      - run:
+          working_directory: ~/expo/android
+          command: zipalign -v -p 4 ./app/build/outputs/apk/prodMinSdkProdKernel/release/app-prodMinSdk-prodKernel-release-unsigned.apk app-prod-release-unsigned-aligned.apk
```

Comment:
write out the flags in long form (`--`) here, maybe use a multline yaml string

Reply:
I agree with that idea, but they don't seem to have a long form:
```
$ docker run -it --rm circleci/android:api-27-ndk-r17b /opt/android/sdk/build-tool
s/27.0.3/zipalign -h
Zip alignment utility
Copyright (C) 2009 The Android Open Source Project

Usage: zipalign [-f] [-p] [-v] [-z] <align> infile.zip outfile.zip
       zipalign -c [-p] [-v] <align> infile.zip

  <align>: alignment in bytes, e.g. '4' provides 32-bit alignment
  -c: check alignment only (does not modify file)
  -f: overwrite existing outfile.zip
  -p: memory page alignment for stored shared object files
  -v: verbose output
  -z: recompress using Zopfli
```